hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +61,106 @@
+                .filter(HotRestartConfig::isEnabled).count();
     }
 
     private long countMapWithWANReplication() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getWanReplicationRef() != null).count();
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
+        return p -> p.getMapStoreConfig().isEnabled();
+    }
+
+    static Predicate<MapConfig> isMapStoreDisabled() {
+        return p -> !p.getMapStoreConfig().isEnabled();
+    }
+
+    static class LatencyInfo {
+        private static final LatencyInfo ZERO = new LatencyInfo(0, 0);
+        private final long totalLatency;
+        private final long totalOperation;
+
+        LatencyInfo(long totalLatency, long totalOperation) {
+            this.totalLatency = totalLatency;
+            this.totalOperation = totalOperation;
+        }
+
+        LatencyInfo sum(LatencyInfo latencyInfo) {
+            return new LatencyInfo(this.totalLatency + latencyInfo.totalLatency, this.totalOperation + latencyInfo.totalOperation);
+        }
+    }
+
+
+    private long mapPutLatency(Node node, Predicate<MapConfig> predicate) {
+
+        LatencyInfo latencyInfo = Optional.ofNullable(mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getPutOperationCount() != 0L)
+                .map(mapStats -> new LatencyInfo(mapStats.getTotalPutLatency(), mapStats.getPutOperationCount()))
+                .reduce(LatencyInfo.ZERO, LatencyInfo::sum)).orElse(LatencyInfo.ZERO);
```

Comment:
This is actually very nicely done. I'm worried about the performance though. This will create a new `LatencyInfo` object for every maps (with mapstore) , and that just generates a little bit too much garbage for the GC. In some rare cases the user can have 100k+ maps (which doesn't make much sense but in some crazy edge cases it happens). So it would be better to accumulate the values into a single (mutable) `LatencyInfo` object (which isn't a big change actually). 


Thumbs up for applying the null object pattern and immutability. It would be very nice OO design under regular circumstances.